enclave_manager/avalon_enclave_manager/wpe/wpe_requester.py

```diff
@@ -41,73 +46,157 @@ def __init__(self, config):
+                decrypted_res = crypto_utils.decrypted_response(
+                    wo_response_json, session_key, session_iv)
+                return decrypted_res
+            return None
         else:
             logger.error("Could not register this WPE with the KME : {}"
                          .format(response))
             return None
 
-    def preprocess_work_order(self, wo_request, encryption_key):
+    def preprocess_work_order(self, wo_request, encryption_key,
+                              verifying_key):
         """
         Request to preprocess a work order
 
         Parameters :
             @param wo_request - The original work order request
             @param encryption_key - WPE's public encryption key
+            @param verifying_key - Verifying key of the worker
         Returns :
             @returns result - Result from KME that includes the workorder
                               key info. None, in case of failure.
         """
+        # @TODO : Need to construct a full work order request as with
+        # uid and reg requests. Generating the session key here and
+        # verifying after response is received.
         json_rpc_request = self._get_request_json("PreProcessWorkOrder")
-        json_rpc_request["params"] = {
-            "wo_request": wo_request,
-            "encryption_key": encryption_key
-        }
+        json_rpc_request["params"] = {"wo_request": wo_request,
+                                      "encryption_key": encryption_key}
         response = self._post_and_get_result(json_rpc_request)
 
-        if "result" in response:
-            return response["result"]
-        else:
-            logger.error("The KME could not preprocess the work order: {}"
-                         .format(response))
-            return None
+        # @TODO : Response handling to be implemented
+        return None
+
+    def _construct_wo_req(self, in_data, workload_id, encryption_key,
+                          session_key, session_iv):
+        """
+        Construct the parameters for a standard work order request
+
+        Parameters :
+            @param in_data - In data to be passed to workload processor
+            @param workload_id - Id of the target workload
+            @encryption_key - Worker encryption key
+            @session_key - Session key to be embedded in request
+            @session_iv - Session key iv for encryption algorithm
+        Returns :
+            @returns A json request prepared using parameters passed
+        """
+        # Create work order
+        # Convert workloadId to hex
+        workload_id_hex = workload_id.encode("UTF-8").hex()
+        work_order_id = secrets.token_hex(32)
+        requester_id = secrets.token_hex(32)
+        requester_nonce = secrets.token_hex(16)
+        # worker id is not known here. Hence passing a random string
+        worker_id = secrets.token_hex(32)
```

Comment:
worker ID should be ID of KME worker corresponding to this WPE. Please read it from config file

Reply:
Added a commit.